src/Domain/Person.php

```diff
@@ -4,24 +4,62 @@
 
 namespace BigGive\Identity\Domain;
 
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
 use JsonSerializable;
 
 /**
- * @todo ORM map everything!
+ * @ORM\Entity()
+ * @ORM\HasLifecycleCallbacks
+ * @ORM\Table
  */
 class Person implements JsonSerializable
 {
-    public ?int $id = null; // TODO decide between numeric and UUIDs as primary.
+    use TimestampsTrait;
 
+    /**
+     * @ORM\OneToMany(targetEntity="PaymentMethod", mappedBy="person", fetch="EAGER")
+     * @var Collection|PaymentMethod[]
+     */
+    public Collection | array $paymentMethods;
+
+    // TODO decide between numeric and UUIDs as primary/ whether UUIDs worth it at all.
```

Comment:
@NoelLH What's the pros and cons of each? I'd be interested in knowing

Reply:
@alihejazi I think this has changed a bit over time and also depends on specific database systems (DBMS's), their versions and their engines.

To my mind when we were doing this for MatchBot, specifically donations which are a similar object to people & payment methods here in that they are somewhat sensitive but also need good performance and ability to scale, some factors for UUIDs were:

* **Pro**: race conditions are never likely to be an issue and multiple apps can generate and "know" an objects identifier up-front, without waiting for the DB or any locks; this remains no true no matter how much apps horizontally scale. Auto-incrementing doesn't have this because either:
  1. the client asserts / tries to claim an ID, in which case it can break with collisions, or
  2. the client waits for the DB's auto-increment response and so cannot use any identifier for logs, responses etc. until later in the object creation process
* **Pro**: pseudo-randomess gives possible bad actors less transparent information about the scale of our data and less routes into guessing IDs that could help them attack real users. This is a good reason to _have_ UUIDs and make them the main thing an API exposes, but not particularly a reason they should be a primary identifier.
* **Con**: performance as MySQL primary keys. Until recently this was basically always worse than auto increment IDs. This might be changing now if you choose good configurations.
* **Con**: an extra library mapping the type for Doctrine means one more thing that could go wrong, even if it is well maintained + flexible
* **Not-really Con**: more storage/network bytes in most implementations – but this is basically a non-issue in modern infrastructure and compared to all the big strings we need.

So, refreshing my memory writing this and looking at the code comments it's pretty obvious how MatchBot's `Donation` model ended up as it is:
* an auto-increment primary key (for every single `Model` in the app, in fact)
* _and_ an auto-generated UUID for API back and forth.

I think copying this would be a safe enough bet. However there is also maybe a slightly more radical option open to us now. **Binary** UUIDs as a MySQL type have native helper fn support as of 8.0, and the lib we use already for UUIDs has a type you can register that is binary backed an additionally a [version that claims to be a performant choice to InnoDB primary keys](https://github.com/ramsey/uuid-doctrine#innodb-optimised-binary-uuids).

As the app is brand new, maybe it would be interesting to try out using UUIDs as our primary keys and see if we find any downsides?

Reply:
@NoelLH Thanks for the fantastic overview! Makes a lot of sense. I'm leaning towards UUIDs, and am happy to try them out as you've suggested when I have some time.